diff --git a/src/draw/lv_draw.c b/src/draw/lv_draw.c
index dc4ce2219..f0787814a 100644
--- a/src/draw/lv_draw.c
+++ b/src/draw/lv_draw.c
@@ -521,7 +521,8 @@ void * lv_draw_layer_alloc_buf(lv_layer_t * layer)
 
 void * lv_draw_layer_go_to_xy(lv_layer_t * layer, int32_t x, int32_t y)
 {
-    return lv_draw_buf_goto_xy(layer->draw_buf, x, y);
+    if(x < 0 || y < 0) return NULL;
+    return lv_draw_buf_goto_xy(layer->draw_buf, (uint32_t)x, (uint32_t)y);
 }
 
 lv_draw_task_type_t lv_draw_task_get_type(const lv_draw_task_t * t)
diff --git a/src/draw/lv_draw_buf.c b/src/draw/lv_draw_buf.c
index 293f23d5e..fa1116744 100644
--- a/src/draw/lv_draw_buf.c
+++ b/src/draw/lv_draw_buf.c
@@ -378,7 +378,7 @@ void * lv_draw_buf_goto_xy(const lv_draw_buf_t * buf, uint32_t x, uint32_t y)
     if(buf == NULL) return NULL;
 
     if(x >= buf->header.w || y >= buf->header.h) {
-        LV_LOG_ERROR("coordinates out of range, x: %" LV_PRIu32 ", y: %"LV_PRIu32", w: %"LV_PRIu32", h: %"LV_PRIu32, x, y,
+        LV_LOG_WARN("coordinates out of range, x: %" LV_PRIu32 ", y: %"LV_PRIu32", w: %"LV_PRIu32", h: %"LV_PRIu32, x, y,
                      (uint32_t)buf->header.w, (uint32_t)buf->header.h);
         return NULL;
     }
diff --git a/src/draw/sw/lv_draw_sw_mask_rect.c b/src/draw/sw/lv_draw_sw_mask_rect.c
index 7abb8a3ed..a43a6f4cb 100644
--- a/src/draw/sw/lv_draw_sw_mask_rect.c
+++ b/src/draw/sw/lv_draw_sw_mask_rect.c
@@ -97,6 +97,7 @@ void lv_draw_sw_mask_rect(lv_draw_task_t * t, const lv_draw_mask_rect_dsc_t * ds
 
         lv_color32_t * c32_buf = lv_draw_layer_go_to_xy(target_layer, draw_area.x1 - buf_area->x1,
                                                         y - buf_area->y1);
+        if(c32_buf == NULL) continue;
 
         if(res == LV_DRAW_SW_MASK_RES_TRANSP) {
             lv_memzero(c32_buf, area_w * sizeof(lv_color32_t));
